src/core/models/answersconfig.js

```diff
@@ -0,0 +1,216 @@
+     * @type {boolean}
+     */
+    this.useTemplates = parsedConfig.useTemplates;
+
+    /**
+     * Optional, additional templates to register with the renderer
+     * @type {Object}
+     */
+    this.templateBundle = parsedConfig.templateBundle;
+
+    /**
+     * Optional, provide configuration for each vertical that is shared across components
+     * @type {VerticalPagesConfig}
+     */
+    this.verticalPages = new VerticalPagesConfig(parsedConfig.verticalPages);
+
+    /**
+     * Optional, search specific settings
+     * @type {SearchConfig}
+     */
+    this.search = new SearchConfig(parsedConfig.search);
+
+    /**
+     * Optional, vertical no results settings. See Vertical No Results
+     * @type {Object}
+     */
+    this.noResults = parsedConfig.noResults;
+
+    /**
+     * Optional, the locale will affect how queries are interpreted and the results returned. Defaults to 'en'
+     * @type {string}
+     */
+    this.locale = parsedConfig.locale;
+
+    /**
+     * Optional, the Answers Experience version to use for api requests
+     * @type {string}
+     */
+    this.experienceVersion = parsedConfig.experienceVersion;
+
+    /**
+     * Optional, prints full Answers error details when set to \`true\`. Defaults to false.
+     * @type {boolean}
+     */
+    this.debug = parsedConfig.debug;
+
+    /**
+     * Optional, If true, the search session is tracked. If false, there is no tracking. Defaults to true.
+     * @type {boolean}
+     */
+    this.sessionTrackingEnabled = parsedConfig.sessionTrackingEnabled;
+
+    /**
+     * Optional, invoked when the state of any component changes
+     * @type {function()}
+     */
+    this.onStateChange = parsedConfig.onStateChange;
+
+    /**
+     * Optional, analytics callback after a vertical search, see onVerticalSearch Configuration for additional details
+     * @type {function()}
+     */
+    this.onVerticalSearch = parsedConfig.onVerticalSearch;
+
+    /**
+     * Optional, analytics callback after a universal search, see onUniversalSearch Configuration for additional details
+     * @type {function()}
+     */
+    this.onUniversalSearch = parsedConfig.onUniversalSearch;
+
+    /**
+     * Optional, opt-out of automatic css variable resolution on init for legacy browsers
+     * @type {boolean}
+     */
+    this.disableCssVariablesPonyfill = parsedConfig.disableCssVariablesPonyfill;
+
+    /**
+     * Optional, the analytics key describing the Answers integration type. Accepts 'STANDARD' or 'OVERLAY', defaults to 'STANDARD'
+     * @type {string}
+     */
+    this.querySource = parsedConfig.querySource;
+
+    this.mock = parsedConfig.mock;
+    this.analyticsOptions = parsedConfig.analyticsOptions;
+    this.fieldFormatters = parsedConfig.fieldFormatters;
+
+    this.validate();
+    Object.freeze(this);
+  }
+
+  /**
+   * Validates the Answers config object to ensure things like api key and experience key are
+   * properly set.
+   */
+  validate () {
```

Comment:
Since this method is only used internally, we would typically prefix it with an underscore